src/fides/api/models/privacy_preference.py

```diff
@@ -388,6 +388,8 @@ def __tablename__(self) -> str:
         MutableDict.as_mutable(JSONB)
     )  # Dict of TCF attributes saved, for a TCF notice
 
+    property_id = Column(String, index=True, nullable=True)
```

Comment:
hm, this is the same as `plus_property.id` right? Just trying to think through if this should be a FK. We are enforcing other resources like the privacy notice, privacy notice history, etc. be proper FK's.  

Cons are that this is a large table for some customers, adding a FK is tricky.  It could be added with something like a "NOT VALID" at first and then issuing a separate query to validate the constraint separately.

It might not be worth that though, a string could be fine too, just wanted to surface

Reply:
Yes! Same thing, that was my intention but I forgot to add the foreign key constraint 😵‍💫. I set it to be nullable since it's not guaranteed that the associated privacy experience will be retrieved by property ID, which is the main bit of info we're tracking with this new addition. I also set `ondelete="SET NULL"` since we want to be able to delete properties freely and still keep these historical records around.

I didn't know about "NOT VALID" in Postgres so thanks for sharing! How would that apply here? From my understanding it allows us to delay the constraint validation and create the constraint more quickly. Would this optimization still be needed if every row has a null `property_id`?

Reply:
discussed in person, we'll revert back to a string to preserve the history in case the property is deleted in the future.